### PR TITLE
Add sample: Front Door (Standard/Premium) with rule set

### PR DESCRIPTION
**Pull request by johndowns**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add sample: Front Door (Standard/Premium) with rule set

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

File: 201-front-door-standard-premium-rule-set/prereqs/prereq.azuredeploy.bicep [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2019-06-01' = {
+  name: accountName
+  location: location
+  kind: 'StorageV2'
+  sku: {
+    name: skuName
+  }
+  properties: {
+    supportsHttpsTrafficOnly: false // This is only configured to make this sample work correctly with Application Gateway. This is not recommended practice - for production solutions you should always use end-to-end HTTPS.
```
Comment (bmoore-msft) [inline 201-front-door-standard-premium-rule-set/prereqs/prereq.azuredeploy.bicep:36]: why can't we set it to true?  and show users how to do that?

Comment (johndowns) [reply 201-front-door-standard-premium-rule-set/prereqs/prereq.azuredeploy.bicep:36]: Sorry - that was a holdover from an earlier version of the module. I've removed it.

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Pull request merged**